src/server/AccountsServer.js

```diff
@@ -15,17 +27,38 @@ class Accounts extends AccountsCommon {
     }
     this.db = db;
   }
+  async createUser(user: UserCreationInputType): Promise<string> {
+    if (this.validateUsername(user.username, false) && this.validateEmail(user.email, false)) {
+      throw new AccountsError({ message: 'Username or Email is required' });
+    }
+    if (user.username && await this.db.findUserByUsername(user.username)) {
```

Comment:
I think a better approach here would be to let the `dbInterface` have a general `findUser` method that will accept `username`, `email`, `id`, `loginToken` or whatever identifier. This way the logic for how the data store holds his users and finds them is not tied to this `createUser` logic. 

Some `dbInterfaces` might be able to do this check with only one round trip to the actual db.